### DISCUSSION TITLE
Problem with so-allow on heavynode

### DISCUSSION THREAD

**Discussion by n0th1nt0lOOz** [Security-Onion-Solutions/securityonion#5697] [Unsupported Versions] [answered]
Hi all
Sorry for asking a lot

I have SO 2.3.70 with distributed mode
so-allow on my manager is ok. but on my heavynode they throw error. I try to test with the so-firewall

```
[root@chnode local]# so-firewall listhostgroups
Traceback (most recent call last):
  File "/sbin/so-firewall", line 392, in <module>
    main()
  File "/sbin/so-firewall", line 382, in main
    code = cmd(options, args[1:])
  File "/sbin/so-firewall", line 285, in listhostgroups
    return listHostGroups()
  File "/sbin/so-firewall", line 81, in listHostGroups
    content = loadYaml(hostgroupsFilename)
  File "/sbin/so-firewall", line 73, in loadYaml
    file = open(filename, "r")
FileNotFoundError: [Errno 2] No such file or directory: '/opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml'
```
I try to add beats host on manager and salt highstate but maybe it's not the right way. In this case my heavy node is missing something right?

**Comment by n0th1nt0lOOz** [comment, marked as answer]
Waiting for help :D

**Reply by TOoSmOotH** [reply]
From searching the word firewall on the docs you will see:

Managing firewall rules, for all devices, should be done from the manager node using either so-allow, so-firewall or, for advanced cases, manually editing the yaml files.

**Reply by n0th1nt0lOOz** [reply]
@TOoSmOotH yes I tried and know that. But in distributed mode, winlogbeat or filebeat send log to heavynode or manager node? because when I point winlogbeat logstash output to heavynode and it show connection refuse.
Thanks
